src/main/java/seedu/loyaltylift/logic/commands/EditOrderCommand.java

```diff
@@ -0,0 +1,192 @@
+    public EditOrderCommand(Index index, EditOrderDescriptor editOrderDescriptor) {
+        requireNonNull(index);
+        requireNonNull(editOrderDescriptor);
+
+        this.index = index;
+        this.editOrderDescriptor = new EditOrderDescriptor(editOrderDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Order> lastShownList = model.getFilteredOrderList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_ORDER_DISPLAYED_INDEX);
+        }
+
+        Order orderToEdit = lastShownList.get(index.getZeroBased());
+        Order editedOrder = createEditedOrder(orderToEdit, editOrderDescriptor);
+
+        model.setOrder(orderToEdit, editedOrder);
+        model.updateFilteredOrderList(PREDICATE_SHOW_ALL_ORDERS);
+        return new CommandResult(String.format(MESSAGE_EDIT_ORDER_SUCCESS, editedOrder));
+    }
+
+    /**
+     * Creates and returns a {@code Order} with the details of {@code orderToEdit}
+     * edited with {@code editOrderDescriptor}.
+     */
+    private static Order createEditedOrder(
+            Order orderToEdit, EditOrderDescriptor editOrderDescriptor) {
+        assert orderToEdit != null;
+
+        Name updatedName = editOrderDescriptor.getName().orElse(orderToEdit.getName());
+        Status currentStatus = orderToEdit.getStatus();
+        Quantity updatedQuantity = editOrderDescriptor.getQuantity().orElse(orderToEdit.getQuantity());
+        Address updatedAddress = editOrderDescriptor.getAddress().orElse(orderToEdit.getAddress());
+
+        return new Order(updatedName, updatedQuantity, currentStatus, updatedAddress);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        // short circuit if same object
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof EditOrderCommand)) {
+            return false;
+        }
+
+        // state check
+        EditOrderCommand e = (EditOrderCommand) other;
+        return index.equals(e.index)
+                && editOrderDescriptor.equals(e.editOrderDescriptor);
+    }
+
+    /**
+     * Stores the details to edit the order with. Each non-empty field value will replace the
+     * corresponding field value of the order.
+     */
+    public static class EditOrderDescriptor {
+        private Name name;
+        private Quantity quantity;
+        private Address address;
+        private Status status;
+
+        public EditOrderDescriptor() {}
+
+        /**
+         * Copy constructor.
+         * A defensive copy of {@code tags} is used internally.
+         */
+        public EditOrderDescriptor(EditOrderDescriptor toCopy) {
+            setName(toCopy.name);
+            setQuantity(toCopy.quantity);
+            setAddress(toCopy.address);
+            this.status = toCopy.status;
```

Comment:
Maybe we can consider not having the `status` attribute in `EditOrderDescriptor`. Since the edit command does not allow a change in status, the Order's status attribute should always remain the same. Also, the status attribute on initialisation is always null based on its current usages. Thus, the status attribute in the descriptor feels redundant.